### DISCUSSION TITLE
Feature request: Migrations

### DISCUSSION THREAD

**Discussion by filidorwiese** [MaterializeInc/materialize#14212] [Technical musings]
Hi! Materialize looks awesome! We're considering moving away from KSQL to Materialize for features like self-joins and Postgres driver compatibility. However, the one thing that is preventing us from adopting Materialize is some form of migrations tooling.

I probably won't have to explain to database developers the merits of migrations to handle schema updates on production environments. Could you shed some light on how people are currently using Materialize without it? Are there some plans to create a migrations tool in the near future? Or do people roll their own?

For sake of completeness, this is a link to the [KSQL migration tool](https://docs.ksqldb.io/en/latest/operate-and-deploy/migrations-tool/) documentation, which in our experience does the job adequately.

**Comment by bobbyiliev** [comment]
Hi there @filidorwiese 👋 

One option to do this with Materialize is with dbt. Here is link to the official documentation on how to use dbt to manage Materialize:

- [How to use dbt to manage Materialize](https://materialize.com/docs/integrations/dbt/)

Also, here is another helpful link for a quick introduction:

- [Introducing: dbt + Materialize](https://materialize.com/introducing-dbt-materialize/)

I also have this demo that shows this in action together with Redpanda:

- [Demo: How to use dbt with Materialize](https://github.com/bobbyiliev/materialize-tutorials/tree/main/mz-user-reviews-dbt-demo)

Depending on your usecase, another option might be to use a side container that you could use to execute a startup SQL script to create your schema, here is a quick Docker-compose example with this in action:

- [Materialize bootstrap SQL script example](https://github.com/bobbyiliev/materialize-tutorials/tree/main/mz-bootstrap-example)

Another option that I've tried in the past is to use a migration tool like the following, though the level of support for such migration tools might be limited at the moment but will improve as we extend the coverage of `pg_catalog` in Materialize (#2157):

- [Laravel database migrations with Materialize](https://github.com/bobbyiliev/mzschema)

Hope that this helps!

Best,

Bobby

**Reply by filidorwiese** [reply]
Thank you, I'll look into those options
